Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -531,20 +542,24 @@ extension OrderStore {
 
     /// Updates the Status of the specified Order, as requested.
     ///
-    /// - Returns: Status, prior to performing the Update OP.
-    ///
-    @discardableResult
-    func updateOrderStatus(siteID: Int64, orderID: Int64, statusKey: OrderStatusEnum) -> OrderStatusEnum {
-        let storage = storageManager.viewStorage
-        guard let order = storage.loadOrder(siteID: siteID, orderID: orderID) else {
-            return statusKey
-        }
-
-        let oldStatus = order.statusKey
-        order.statusKey = statusKey.rawValue
-        storage.saveIfNeeded()
-
-        return OrderStatusEnum(rawValue: oldStatus)
+    func updateOrderStatus(siteID: Int64, orderID: Int64,
+                           statusKey: OrderStatusEnum,
+                           onCompletion: @escaping (_ previousStatus: OrderStatusEnum) -> Void) {
+        storageManager.performAndSave({ storage -> OrderStatusEnum in
+            guard let order = storage.loadOrder(siteID: siteID, orderID: orderID) else {
+                return statusKey
+            }
+            let oldStatus = order.statusKey
+            order.statusKey = statusKey.rawValue
+            return OrderStatusEnum(rawValue: oldStatus)
+        }, completion: { result in
+            switch result {
+            case .success(let status):
+                onCompletion(status)
+            case .failure: // this case should not happen
```

Comment:
Just wondering here: This would happen if a plugin adds a custom order status, right? (or maybe would be catched by the custom case 🤔 ). Perhaps we should add a DDLog just for easiness of future debugging?

Reply:
This doesn't happen because no error is thrown in the implementation of the operation closure. I updated the comment in 09fd7ba4e796178f52fbc20932d0ec816614b14d for better clarity.